R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+#' paste3
+#'
+#' An internal fucntion that removes NAs from paste.
+#' Source: https://stackoverflow.com/users/1855677/42.
+#'
+#' @param ... objects to be pasted together.
+#' @param sep separating string.
+
+paste3 <- function(..., sep = ", ") {
+  L <- list(...)
+  L <- lapply(L, function(x) {
+    x[is.na(x)] <- ""
+    x
+  })
+  ret <- gsub(
+    paste0("(^", sep, "|", sep, "$)"), "",
+    gsub(
+      paste0(sep, sep), sep,
+      do.call(paste, c(L, list(sep = sep)))
+    )
+  )
+  is.na(ret) <- ret == ""
+  ret
+}
+
+#' Supply an expression to identify which header groups are hooked.
+#'
+#' @description
+#' This function is used with the `.hook_if` or `.hook_if_rev` arguments in the `locate_groups` function.
+#' It passes an expression to `dplyr::summarise` that identifies which header groups are hooked - for example swiched from N to NNW.
+#' See the the `locate_groups` documentation for more information and an example.
+#'
+#' @param ...  expression applied to a  identifies which header groups are hooked
+#'
+#' @name get_range_dfs
+#' @export
+
+hook <- function(...) {
+  rlang::quos(...)
+}
+
+#' Unmerge and fill in blanks for row headers.
+#'
+#' This is an internal function that function ensures that merged cells are unmerged.
+#'
+#' @param header_df a data frame containing header cells.
+#' @param header_fill the criteria according to which blank cells will be filled.
+#' @param formats the formats of the workbook associated with the tidyxl data frame.
+
+fill_blanks_in_row_headers <- function(header_df, header_fill, formats) {
+  if (header_fill == "style") {
+    continue <- TRUE
+    
+    while (continue) {
+      sheet_original <- header_df
+      header_df <- header_df %>% unmerge_row_cells(strict_merging = FALSE)
+      
+      continue <- !identical(sheet_original, header_df)
+    }
```

Comment:
How well does this perform?  Would it be possible to unmerge in one go?  I realise I still haven't fulfilled my promise to do the unmerging in tidyxl.

Reply:
This can produce some slight delay - no longer than a second in my experience. I'll need to give more thought to how it might be done in one go.